background/providers/torrentproject.js

```diff
@@ -0,0 +1,99 @@
+var https = require('https');
+var zlib = require('zlib');
+var path = require('path');
+var nconf = require('nconf');
+var mongoose = require('mongoose');
+var Category = require('../../app/models/Category.js');
+var Torrent = require('../../app/models/Torrent.js');
+
+nconf.use('memory');
+nconf.argv().env('__').file({
+    file: path.resolve(__dirname + '/../../config.json')
+});
+
+if(nconf.get('database:mongodb:enabled')){
+    mongoose.connect('mongodb://' + nconf.get('database:mongodb:host') + ':' + nconf.get('database:mongodb:port') + '/' + nconf.get('database:mongodb:collection'), function(err){
```

Comment:
You're loading the files into a file with a connection already established. We can just remove that from the providers and keep it in the import file so we have a single connection shared between all providers.